samples/net/gsm_modem/src/main.c

```diff
@@ -26,18 +70,23 @@ static void event_handler(struct net_mgmt_event_callback *cb,
 
 	if (mgmt_event == NET_EVENT_L4_CONNECTED) {
 		LOG_INF("Network connected");
+		connected = true;
 		return;
 	}
 
 	if (mgmt_event == NET_EVENT_L4_DISCONNECTED) {
 		LOG_INF("Network disconnected");
+		connected = false;
 		return;
 	}
 }
 
 int main(void)
 {
-	const struct device *uart_dev = device_get_binding(CONFIG_MODEM_GSM_UART_NAME);
+	const struct device *uart_dev =
+				device_get_binding(CONFIG_MODEM_GSM_UART_NAME);
+
+	gsm_dev = device_get_binding("modem_gsm");
```

Comment:
I am not fond of using the hard coded value here. Could you take the commit from my PR that uses a define.